caom2/src/main/java/ca/nrc/cadc/caom2/types/MultiPolygon.java

```diff
@@ -0,0 +1,367 @@
+*  products derived from this           promouvoir les produits dérivés
+*  software without specific prior      de ce logiciel sans autorisation
+*  written permission.                  préalable et particulière
+*                                       par écrit.
+*
+*  This file is part of the             Ce fichier fait partie du projet
+*  OpenCADC project.                    OpenCADC.
+*
+*  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+*  you can redistribute it and/or       vous pouvez le redistribuer ou le
+*  modify it under the terms of         modifier suivant les termes de
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+*  $Revision: 5 $
+*
+************************************************************************
+*/
+
+package ca.nrc.cadc.caom2.types;
+
+import ca.nrc.cadc.util.HexUtil;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author pdowler
+ */
+public class MultiPolygon implements Serializable
+{
+    private static final long serialVersionUID = 201708141600L;
+    
+    private final List<Vertex> vertices = new ArrayList<Vertex>();
+
+    // lazily computed
+    private transient Point center;
+    private transient Double area;
+    private transient Circle minimumSpanningCircle;
+    private transient Boolean ccw;
+
+    public MultiPolygon() { }
+
+    private void validate()
+    {
+        if (vertices.size() < 4) // triangle
+            throw new IllegalPolygonException("invalid polygon: " + vertices.size() + " vertices");
+
```

Comment:
Isn't the pseudo-alg:
openLoop = false;
for vertices
{
    if vertices.type is MOVE
   {
       if openLoop {throw exception("MOVE when loop open")}
       openLoop = true
   }
   else if vertices.type is CLOSE
   {
      if !openLoop{ throw exception("CLOSE when loop closed")}
      openLoop = false
   }
   else
   {  
       // LINE
      if !openLoop { throw exception("LINE when loop close"}
}